ecosystem/typescript/sdk/examples/typescript/multisig_managed_coin.ts

```diff
@@ -0,0 +1,294 @@
+    arguments: [owner1.address().hex()],
+  };
+  const multisigAddress = (await client.view(payload))[0] as string;
+
+  const createMultisigManagedCoin = await client.generateTransaction(owner1.address(), {
+    function: MODULE_ADDR + "::multisig_managed_coin::initialize",
+    type_arguments: [],
+    arguments: [
+      [owner2.address().hex(), owner3.address().hex()],
+      2,
+      ["description"],
+      [BCS.bcsSerializeStr("The multisig account for Meme")],
+      0,
+      "meme coin",
+      ASSET_SYMBOL,
+      8,
+      "http://meme.xyz/favicon.ico",
+      "http://meme.xyz",
+      [true, true, true],
+    ],
+  });
+  // <:!:section_2
+  await client.generateSignSubmitWaitForTransaction(owner1, createMultisigManagedCoin.payload);
+  assert((await getSignatureThreshold(client, multisigAddress)) == 2);
+  assert((await getNumberOfOwners(client, multisigAddress)) == 3);
+
+  // Fund the multisig account for gas.
+  await faucetClient.fundAccount(multisigAddress, 100_000_000);
+
+  // Step 2: Create a multisig transaction to mint coins to an account.
+  // We'll be including the full payload to be stored on chain.
+  // ===========================================================================================
+
+  // Deterministically calculate the named metadata object address from the creator address and asset symbol.
+  const metadata = getNamedObjectAddress(owner1, ASSET_SYMBOL);
+  console.log(`metadata: ${metadata.toHexString()}`);
+
+  // Create the mint transaction payload
+  {
+    // :!:>section_3
+    const recipientsSerializer = new BCS.Serializer();
+    BCS.serializeVector(
+      [AccountAddress.fromHex(owner2.address()), AccountAddress.fromHex(owner3.address())],
+      recipientsSerializer,
+    );
+    const mintTxPayload = new MultiSigTransactionPayload(
+      EntryFunction.natural(
+        MODULE_ADDR + "::managed_fungible_asset",
+        "mint_to_primary_stores",
+        [],
+        [
+          BCS.bcsToBytes(metadata),
+          recipientsSerializer.getBytes(),
+          BCS.serializeVectorWithFunc([1_000, 2_000], "serializeU64"),
+        ],
+      ),
+    );
+    const mintTxExecution = new TransactionPayloadMultisig(new MultiSig(AccountAddress.fromHex(multisigAddress)));
+
+    // Create the mint multisig tx on chain.
+    const mintTx = await client.generateTransaction(owner2.address(), {
+      function: "0x1::multisig_account::create_transaction",
+      type_arguments: [],
+      arguments: [multisigAddress, BCS.bcsToBytes(mintTxPayload)],
+    });
+    await client.generateSignSubmitWaitForTransaction(owner2, mintTx.payload);
+
+    // Owner 3 approves.
+    await approve(client, owner3, multisigAddress, 1);
+
+    // Owner 2 can now execute the transactions as it already has 2 approvals (from owners 2 and 3).
+    await client.generateSignSubmitWaitForTransaction(owner2, mintTxExecution);
+    // <:!:section_3
+    // Check the primary store balance of owner2 and owner3.
```

Comment:
wonder if can use `FungibleAssetClient` https://github.com/aptos-labs/aptos-core/blob/main/ecosystem/typescript/sdk/src/plugins/fungible_asset_client.ts#L59

Reply:
I don't quite get why FAclient has to depends on `Provider` which needs a indexer client too. FA client has nothing to do with indexer, right?